From 4408178b9510e26ef1f9e81484dc5a0a5125e6bf Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Thu, 12 Nov 2015 13:20:08 +0100
Subject: [PATCH 094/117] sparsify: read all 'qemu-img --help' output
 (RHBZ#1218934)

It seems like qemu-img might sometimes not like when all of its output
for --help is not read fully; while this makes us read more data from it
than currently (only the first line), it should make qemu-img more
stable.
---
 sparsify/sparsify.ml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/sparsify/sparsify.ml b/sparsify/sparsify.ml
index 2475bb0..de35a11 100644
--- a/sparsify/sparsify.ml
+++ b/sparsify/sparsify.ml
@@ -181,7 +181,10 @@ read the man page virt-sparsify(1).
 let qemu_img_version =
   let cmd = "qemu-img --help" in
   let chan = open_process_in cmd in
-  let line = input_line chan in
+  let lines = ref [] in
+  (try while true do lines := input_line chan :: !lines done
+   with End_of_file -> ());
+  let lines = List.rev !lines in
   let stat = close_process_in chan in
   (match stat with
   | WEXITED _ -> ()
@@ -191,6 +194,8 @@ let qemu_img_version =
     error (f_"external command '%s' stopped by signal %d") cmd i
   );
 
+  let line = List.hd lines in
+
   try
     sscanf line "qemu-img version %d.%d" (
       fun major minor ->
-- 
2.5.0

